cmake/mcuboot.cmake

```diff
@@ -10,6 +10,39 @@
 # Since this file is brought in via include(), we do the work in a
 # function to avoid polluting the top-level scope.
 
+function(convert_to_uf2 artifacts slot_addr)
+  if(CONFIG_BUILD_OUTPUT_UF2_USE_FLASH_BASE)
+    set(flash_addr "${CONFIG_FLASH_BASE_ADDRESS}")
+  else()
+    set(flash_addr "${CONFIG_FLASH_LOAD_OFFSET}")
+  endif()
+
+  math(EXPR flash_addr
+      "${flash_addr} + ${slot_addr}"
+       OUTPUT_FORMAT HEXADECIMAL
+  )
+
+  set(uf2_byproducts)
+  foreach(artifact IN LISTS ${artifacts})
+    string(LENGTH ${artifact} len)
+    math(EXPR len "${len} - 3")
+    string(SUBSTRING ${artifact} 0 ${len} output)
+    string(APPEND output "uf2")
+
+    set_property(GLOBAL APPEND PROPERTY extra_post_build_commands
+      COMMAND ${PYTHON_EXECUTABLE} ${ZEPHYR_BASE}/scripts/build/uf2conv.py
+              -c
+              -f ${CONFIG_BUILD_OUTPUT_UF2_FAMILY_ID}
+              -b ${flash_addr}
+              -o ${output}
```

Comment:
Short args are very convenient to users working on command line as it reduces the need for typing, but there are no reasons why you should use short args inside build files.

Please use long args as they are more self-explanatory regarding their meaning.